src/js/components/DataTable/Header.js

```diff
@@ -500,8 +500,9 @@ const Header = forwardRef(
                       ? `${widths[property]}px`
                       : undefined,
                     boxSizing: onResize ? 'border-box' : undefined,
-                    position: onResize ? 'relative' : undefined,
+                    position: onResize && !columnPin ? 'relative' : undefined,
                     overflow: onResize ? 'visible' : undefined,
+                    zIndex: onResize && columnPin ? 2 : undefined,
```

Comment:
Why a value of '2' here?

Reply:
the resizer has z-index: 1;
so I just did one above that.

Reply:
changed!